### ISSUE TITLE
Memory leak when deleting the object on which the modifier is attached

### ISSUE THREAD

**Issue by eliemichel** [eliemichel/OpenMfxForBlender#1]
Steps to reproduce:

 1. Create an object
 2. Attach an OpenMeshEffect modifier to it
 3. Load an OFX plugin in it
 4. Delete the object

See https://devtalk.blender.org/t/freeing-modifiers-runtime-data-on-object-deletion/9314 for details.

**Comment by josephbburg** [comment]
Hi, I built your branch this week and I noticed a similar bug.
Here's my build info:
```
version: 2.80 (sub 74), branch: openmesheffect, commit date: 2019-09-06 21:12, hash: eb6a1a535722, type: 
build date: Tue 09/10/2019, 06:56 PM
platform: Windows
```
And when I open the Blender, add the Open Effects Modifier to the default Cube, then exit, it produces a memory leak error in the console:
```
Error: Not freed memory blocks: 1, total unfreed memory 0.002136 MB
mfx runtime len: 1120 000001AE459CA538
```

Thanks. I hope I can find time to tinker with writing my own plugin sometime. I'm really excited about the future of this project.

Edit:
I think it's the same error, actually, for some reason I thought this bug was already fixed and mine was different! But I was able to reproduce it.
